### ISSUE TITLE
autoimpl trait re-implementation should copy `cfg` attrs

### ISSUE THREAD

**Issue by dhardy** [kas-gui/impl-tools#29]
**Summary**
The `#[cfg(..)]` attr is used to conditionally compile something. If a trait item is conditional, then its re-implementation should be too.

Are there any other attrs which should be forwarded?

**To Reproduce**
```rust
#[impl_tools::autoimpl(for<T: trait> &T)]
trait MyTrait {
    #[cfg(feature = "foo")]
    fn do_foo(&self);
}
```

**Comment by dhardy** [comment]
It seems that no other [built-in attributes](https://doc.rust-lang.org/reference/attributes.html#built-in-attributes-index) should be propegated to method implementations. (There is some ambiguity for `allow`, `warn`, `deny`, `forbid` since something like `#[allow(non_snake_case)]` *can* appear on both trait item declarations and implementations, but testing shows that this is unnecessary.)

**Comment by dhardy** [comment]
Fixed by #30
